src/Microsoft.VisualStudio.Editor.Razor/DefaultVisualStudioDocumentTracker.cs

```diff
@@ -59,11 +59,6 @@ internal class DefaultVisualStudioDocumentTracker : VisualStudioDocumentTracker
                 throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(filePath));
             }
 
-            if (projectPath == null)
```

Comment:
So another approach here would be to not take in the project path but to take in the project path provider. With that approach we'd determine the project path on `Subscribe`. Could be a little nicer but wanted feedback.

Reply:
I'm not sure I understand the difference... what do you see as the tradeoffs?

Reply:
It's not a trade off question but more of a code cleanliness question. You could make the argument about the project changing after unsubscribe => subscribe but that's not a real scenario.

I think I just felt icky about allowing a null parameter in the constructor of the document tracker.

Reply:
It does kinda feel icky. I support the `ProjectPathProvider` approach as long it doesn't involve too much churn.